File: MovieDB-Java/app/src/main/java/com/practice/moviedb/networks/ApiService.java
package com.practice.moviedb.networks;

import com.practice.moviedb.models.TopRatedMovie;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiService {

    @GET("current.json")
    Call<TopRatedMovie> getTopRatedMovies(@Query("api_key") String apiKey,
                                          @Query("language") String language,
                                          @Query("page") String page,
                                          @Query("sort_by") String sortBy);
}

File: MovieDB-Kotlin/app/src/main/java/com/practice/moviedatabase/dal/networks/ApiService.kt
package com.practice.moviedatabase.dal.networks

import com.practice.moviedatabase.models.Genres
import com.practice.moviedatabase.models.TopRatedMovie
import kotlinx.coroutines.Deferred
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiService {

    @GET("movie/top_rated")
    fun getTopRatedMoviesAsync(
        @Query("api_key") apiKey: String,
        @Query("language") language: String,
        @Query("page") page: String,
        @Query("sort_by") sortBy: String
    ): Deferred<TopRatedMovie>

    @GET("genre/movie/list")
    fun getGenresAsync(
        @Query("api_key") apiKey: String,
        @Query("language") language: String
    ): Deferred<Genres>
}
